### PR TITLE
chore: Move styled-icons to devDependencies

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes https://github.com/opencollective/opencollective/issues/1832

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "graphql/named-operations": ["error"],
     "graphql/capitalized-type-name": ["error"],
     "graphql/no-deprecated-fields": ["warn"],
+    "node/no-unpublished-import": ["warn"],
```
Comment (znarf) [inline .eslintrc.json]: Would be great to not have the `warning  "styled-icons" is not published  node/no-unpublished-import`, but also we should not silent this rule. How can we achieve that?

**Comment by znarf** [comment]
Have you tested that it works? In the auto-deployment logs, I see:

```
(node:18) UnhandledPromiseRejectionWarning: Error: Cannot find module 'styled-icons/boxicons-regular/Support'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:611:15)
    at Function.Module._load (internal/modules/cjs/loader.js:537:25)
    at Module.require (internal/modules/cjs/loader.js:665:17)
    at require (internal/modules/cjs/helpers.js:20:18)
...
(node:18) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 72)
```

File: .eslintrc.json (hunk 1 above)
Comment (JohnTitor) [reply .eslintrc.json]: Uhmm, I think https://github.com/opencollective/opencollective/issues/1832#issuecomment-474639542 is a good idea. We have too many errors to control individually.

Comment (JohnTitor) [reply .eslintrc.json]: tl;dr I have no idea

**Comment by JohnTitor** [comment]
We need to investigate more so I'll close for now.

**Pull request closed without merging**